### DISCUSSION TITLE
How to use the getter method with parameters in the public method file outside the component in vue3 XXX.js?

### DISCUSSION THREAD

**Discussion by Abbyzhoum** [vuejs/pinia#1353] [Help and Questions] [answered]
![f8ebb9b2214ad1531f2d63e09a58b0c](https://user-images.githubusercontent.com/38661489/172820250-25e08d43-907c-45d7-a22c-50ef5ab02de2.png)
![abdbe690071778ccef260342d37bb76](https://user-images.githubusercontent.com/38661489/172821058-d845928c-cf3a-40d0-a0ed-e0889c5deff8.png)

When I get the passing arguments to getters in the store in the public js file outside the component, it prompts an error, can anyone help me, thanks

**Comment by posva** [comment, marked as answer]
the type should be `() => CallItem | {}` in the last getter.
BTW you should remove the `get` word from the other getters since they are not functions, just data. Also, `getCallList` isn't needed with pinia, just use `store.calls` directly.

**Reply by Abbyzhoum** [reply]
I understand, thanks
